src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -164,35 +161,11 @@ class FilterButton extends React.Component<Props, State> {
         }
     };
 
-    updateSelectedPrefix = (option: SelectOptionType) => {
-        const prefix = option.displayText;
+    updateSelectedConnector = (option: SelectOptionType) => {
+        const connector = option.displayText;
 
         this.setState({
-            selectedPrefix: prefix,
-        });
-
-        this.updateConditionsPrefixes(prefix);
-    };
-
-    updateConditionsPrefixes = (prefix: string) => {
-        const { conditions } = this.state;
-        const conditionsAfterDeletion = conditions.map((condition, index) => {
-            if (index !== 0) {
-                const updatedCondition = Object.assign({}, condition);
-                updatedCondition.prefix = prefix;
-                return updatedCondition;
-            }
-            return condition;
-        });
-
-        // The first condition must always have a prefix of WHERE.
-        const firstCondition = Object.assign({}, conditionsAfterDeletion[0]);
-        firstCondition.prefix = WHERE;
-
-        const updatedConditions = [firstCondition, ...conditionsAfterDeletion.slice(1)];
-
-        this.setState({
-            conditions: updatedConditions,
+            selectedConnector: connector,
```

Comment:
are there multiple instances of `FilterButton` on a query dropdown? If so, the `selectedConnector` state should be stored in the parent component

Reply:
The query dropdown is within the FilterButton so that technically is the parent component